vizro-core/src/vizro/models/_components/form/dropdown.py

```diff
@@ -212,8 +199,54 @@ def _build_dynamic_placeholder(self):
             _, default_value = get_dict_options_and_default(options=self.options, multi=self.multi)
             self.value = default_value
 
-        return self.__call__(self.options)
+        # The rest of the method is added instead of calling and returning the content from the __call__ method
+        # because placeholder for the Dropdown can't be the dropdown itself. The reason is that the Dropdown value can
+        # be unexpectedly changed when the new options are added. This is developed as the dash feature
+        # https://github.com/plotly/dash/pull/1970.
+        if self.multi:
+            # Add the clientside callback as the callback has to be defined in the page.build process.
+            self._define_clientside_callback()
+            # hidden_select_all_dropdown is needed to ensure that clientside callback doesn't raise the no output error.
+            hidden_select_all_dropdown = [dcc.Dropdown(id=f"{self.id}_select_all", style={"display": "none"})]
+            placeholder_model = dcc.Checklist
+            placeholder_options = self.value
+        else:
+            hidden_select_all_dropdown = [None]
+            placeholder_model = dbc.RadioItems
+            placeholder_options = [self.value]  # type: ignore[assignment]
+
+        description = self.description.build().children if self.description else [None]
+        return html.Div(
+            children=[
+                dbc.Label(
+                    children=[html.Span(id=f"{self.id}_title", children=self.title), *description], html_for=self.id
+                ),
+                placeholder_model(
+                    id=self.id,
+                    options=placeholder_options,
+                    value=self.value,
+                    persistence=True,
+                    persistence_type="session",
+                ),
+                *hidden_select_all_dropdown,
+            ]
+        )
 
     @_log_call
     def build(self):
         return self._build_dynamic_placeholder() if self._dynamic else self.__call__(self.options)
+
+    def _define_clientside_callback(self):
```

Comment:
Shall we call it similar to the function_name it calls?

```suggestion
    def _update_dropdown_select_all"(self):
```